.github/workflows/tests.yml

```diff
@@ -0,0 +1,24 @@
+name: test
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build-and-test:
+    runs-on: windows-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Restore solution
+      run: dotnet restore
+    - name: Build
+      run: dotnet build --configuration Release --no-restore
+    - uses: actions/upload-artifact@v2
+      with:
+        name: workspacer-build
+        path: .\src\workspacer\bin\Release\net5.0-windows\win10-x64
+    - name: Test
+      run: |
+        dotnet tool install -g xunit-cli
+        xunit .\src\workspacer.Shared.Tests\bin\Release\net5.0-windows\win10-x64\workspacer.Shared.Tests.dll .\src\workspacer.Tests\bin\Release\net5.0-windows\win10-x64\workspacer.Tests.dll -verbose
```

Comment:
Any reason to prefer xUnit-cli over plain `dotnet test`?

Reply:
A console runner is needed and xunit-cli wraps around it.

Reply:
Not sure I follow. I use dotnet test with xUnit in my CI jobs without any additional setup. 

Care to elaborate a little further?

Reply:
Oh, for some reason, tests get skipped when I used  `dotnet test` 🤔.
```
  Determining projects to restore...
  All projects are up-to-date for restore.
  workspacer.Shared -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.Shared\bin\Debug\net5.0-windows\win10-x64\workspacer.Shared.dll
  workspacer.FocusIndicator -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.FocusIndicator\bin\Debug\net5.0-windows\win10-x64\workspacer.FocusIndicator.dll
  workspacer.Native -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.Native\bin\Debug\net5.0-windows\win10-x64\workspacer.Native.dll
  workspacer.Bar -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.Bar\bin\Debug\net5.0-windows\win10-x64\workspacer.Bar.dll
  workspacer.Shared.Tests -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.Shared.Tests\bin\Debug\net5.0-windows\win10-x64\workspacer.Shared.Tests.dll
  workspacer.ActionMenu -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.ActionMenu\bin\Debug\net5.0-windows\win10-x64\workspacer.ActionMenu.dll
Test run for C:\Users\sitiom\source\repos\workspacer\src\workspacer.Shared.Tests\bin\Debug\net5.0-windows\win10-x64\workspacer.Shared.Tests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Test Execution Command Line Tool Version 16.8.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
  workspacer.Watcher -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.Watcher\bin\Debug\net5.0-windows\win10-x64\workspacer.Watcher.dll
A total of 1 test files matched the specified pattern.
  workspacer -> C:\Users\sitiom\source\repos\workspacer\src\workspacer\bin\Debug\net5.0-windows\win10-x64\workspacer.dll
  workspacer.Tests -> C:\Users\sitiom\source\repos\workspacer\src\workspacer.Tests\bin\Debug\net5.0-windows\win10-x64\workspacer.Tests.dll
Test run for C:\Users\sitiom\source\repos\workspacer\src\workspacer.Tests\bin\Debug\net5.0-windows\win10-x64\workspacer.Tests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Test Execution Command Line Tool Version 16.8.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     3, Skipped:     0, Total:     3, Duration: 4 ms - workspacer.Shared.Tests.dll (net5.0)

Passed!  - Failed:     0, Passed:     7, Skipped:     0, Total:     7, Duration: 7 ms - workspacer.Tests.dll (net5.0)
```

and xunit-cli fixed it, so I used it:

```
xUnit.net Console Runner v2.4.1 (64-bit .NET 5.0.0)
  Discovering: workspacer.Shared.Tests
  Discovered:  workspacer.Shared.Tests
  Starting:    workspacer.Shared.Tests
    workspacer.Shared.Tests.ColorTests.Creating_Color_With_Invalid_RGB_Components_Throws [STARTING]
    workspacer.Shared.Tests.ColorTests.Creating_Color_With_Invalid_RGB_Components_Throws [FINISHED] Time: 0.0034181s
    workspacer.Shared.Tests.ColorTests.Creating_Color_With_Valid_RGB_Components_Succeeds [STARTING]
    workspacer.Shared.Tests.ColorTests.Creating_Color_With_Valid_RGB_Components_Succeeds [FINISHED] Time: 0.0000679s
    workspacer.Shared.Tests.ColorTests.Different_Colors_Does_Not_Have_Hash_Collision [STARTING]
    workspacer.Shared.Tests.ColorTests.Different_Colors_Does_Not_Have_Hash_Collision [FINISHED] Time: 0.0015202s
  Finished:    workspacer.Shared.Tests
  Discovering: workspacer.Tests
  Discovered:  workspacer.Tests
  Starting:    workspacer.Tests
    workspacer.Tests.SubTests.Does_Not_Have_Hash_Collisions_Between_Different_Keys [STARTING]
    workspacer.Tests.SubTests.Does_Not_Have_Hash_Collisions_Between_Different_Keys [FINISHED] Time: 0.0010873s
    workspacer.Tests.SubTests.LeftCtrl_Does_Not_Equal_RightControl [STARTING]
    workspacer.Tests.SubTests.LeftCtrl_Does_Not_Equal_RightControl [FINISHED] Time: 0.0040807s
    workspacer.Tests.SubTests.LeftCtrl_RightCtrl_And_Ctrl_Has_Same_HashCode [STARTING]
    workspacer.Tests.SubTests.LeftCtrl_RightCtrl_And_Ctrl_Has_Same_HashCode [FINISHED] Time: 0.0002138s
    workspacer.Tests.SubTests.RightAlt_Does_Not_Equal_LeftAlt [STARTING]
    workspacer.Tests.SubTests.RightAlt_Does_Not_Equal_LeftAlt [FINISHED] Time: 0.0000521s
    workspacer.Tests.SubTests.RightAlt_LeftAlt_And_Alt_Has_Same_HashCode [STARTING]
    workspacer.Tests.SubTests.RightAlt_LeftAlt_And_Alt_Has_Same_HashCode [FINISHED] Time: 0.0000338s
    workspacer.Tests.SubTests.RightWin_Does_Not_Equal_LeftWin [STARTING]
    workspacer.Tests.SubTests.RightWin_Does_Not_Equal_LeftWin [FINISHED] Time: 0.0000414s
    workspacer.Tests.SubTests.RightWin_LeftWin_And_WinHas_Same_HashCode [STARTING]
    workspacer.Tests.SubTests.RightWin_LeftWin_And_WinHas_Same_HashCode [FINISHED] Time: 0.00004s
  Finished:    workspacer.Tests
=== TEST EXECUTION SUMMARY ===
   workspacer.Shared.Tests  Total:  3, Errors: 0, Failed: 0, Skipped: 0, Time: 0.068s
   workspacer.Tests         Total:  7, Errors: 0, Failed: 0, Skipped: 0, Time: 0.008s
                                   --          -          -           -        ------
                      GRAND TOTAL: 10          0          0           0        0.076s (0.148s)
```

Reply:
If so, that's something which IMO needs to be fixed too. 

How are you invoking dotnet test? Against main sln-file or something else?

Reply:
> If so, that's something which IMO needs to be fixed too.
> 
> How are you invoking dotnet test? Against main sln-file or something else?

On the sln file.

Reply:
Oh actually nevermind, I didn't read it clearly and it actually passed according to the output above. Thanks for pointing this out.